src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100; // Ensure health is set to 100
+    this.hidden = false; // Initialize hidden to false
+    Animal.alive.push(this); // Add this instance to the alive array
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = !this.hidden; // Invert the hidden property
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    // Ensure we can only bite herbivores and that they aren't hidden
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= 50; // Decrease health by 50
+
+      if (herbivore.health <= 0) {
+        herbivore.die(); // Remove from alive if health is 0 or less
+      }
+    }
```

Comment:
Consider adding a check to ensure the herbivore is alive (i.e., its health is greater than 0) before allowing the carnivore to bite. This will prevent biting an already dead herbivore.